client/app/components/date_range_picker.coffee

```diff
@@ -66,37 +64,26 @@ module.exports = DateRangePicker = React.createClass
         @props.onDateFilter start, end
 
 
-    reset: ->
-        @setState
-            isActive:  false
-            startDate: ''
-            endDate:   '',
-            @filterize
-
-
-    presetYesterday: ->
-        @setState
-            isActive:  true
-            startDate: moment().subtract(1, 'day').format(momentFormat)
-            endDate:   moment().subtract(1, 'day').format(momentFormat),
-            @filterize
+    onReset: ->
+        @changeDates()
 
+    onYesterday: ->
+        @changeDates 'day'
 
-    presetLastWeek: ->
-        @setState
-            isActive:  true
-            startDate: moment().subtract(1, 'week').format(momentFormat)
-            endDate:   moment().format(momentFormat),
-            @filterize
+    onLastWeek: ->
+        @changeDates 'week'
 
+    onLastMonth: ->
+        @changeDates 'month'
 
-    presetLastMonth: ->
-        @setState
-            isActive:  true
-            startDate: moment().subtract(1, 'month').format(momentFormat)
-            endDate:   moment().format(momentFormat),
-            @filterize
+    changeDates: (type) ->
+        value = moment().subtract(1, type).startOf(type) if type
```

Comment:
If we are March 16th I think this will look for message from February 1st to February 29th, where as the previous behaviour was messages from February ~16th to now.

A good middle ground might be messages from February 1st (`stubstract(1).startOf()`) to now.